File: backend/app/schemas/menu.py
import uuid
from datetime import datetime
from pydantic import BaseModel


class MenuTemplateCreate(BaseModel):
    name: str
    category: str
    description: str | None = None
    dish_count: int = 0
    gradient: str = "from-amber-600 to-orange-700"
    dishes: dict | None = None


class MenuTemplateUpdate(BaseModel):
    name: str | None = None
    category: str | None = None
    description: str | None = None
    dish_count: int | None = None
    gradient: str | None = None
    dishes: dict | None = None


class MenuTemplateResponse(BaseModel):
    id: uuid.UUID
    name: str
    category: str
    description: str | None
    dish_count: int
    gradient: str
    dishes: dict | None
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

File: frontend/src/api/menu.ts
import client from './client'

export interface MenuTemplate {
  id: string
  name: string
  category: string
  description: string | null
  dish_count: number
  gradient: string
  dishes: Record<string, unknown> | null
  created_at: string
  updated_at: string
}

export interface MenuTemplateUpdate {
  name?: string
  category?: string
  description?: string | null
  dish_count?: number
  gradient?: string
  dishes?: Record<string, unknown> | null
}

export async function getMenuTemplates(): Promise<MenuTemplate[]> {
  const response = await client.get('/menu-templates')
  return response.data
}

export async function createMenuTemplate(data: MenuTemplateUpdate): Promise<MenuTemplate> {
  const response = await client.post('/menu-templates', data)
  return response.data
}

export async function updateMenuTemplate(id: string, data: MenuTemplateUpdate): Promise<MenuTemplate> {
  const response = await client.put(`/menu-templates/${id}`, data)
  return response.data
}
